src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/MethodBodyScanner.cs

```diff
@@ -319,7 +319,7 @@ public virtual void InterproceduralScan(MethodIL startingMethodBody)
             // are the same set of methods that we discovered and scanned above.
             if (_annotations.CompilerGeneratedState.TryGetCompilerGeneratedCalleesForUserMethod(startingMethod, out List<TypeSystemEntity>? compilerGeneratedCallees))
             {
-                var calleeMethods = compilerGeneratedCallees.OfType<MethodDefinition>();
+                var calleeMethods = compilerGeneratedCallees.OfType<MethodDesc>();
```

Comment:
Since OfType evaluation is lazy, this line is a nop (other than a null check on compilerGeneratedCallees).  Presumably you could comment it out to go along with the subsequent commented out lines?

Reply:
That would work too. I went with this because this file is a line-by-line port from IL Linker where it looks like this:

https://github.com/dotnet/runtime/blob/d92e70f05b8de72758f3a990844e53c76b44d713/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReferenceSource/MethodBodyScanner.cs#L254-L259

(It's identical modulo the odd illinker rule to put a space between method name and parameter list. You can see where the `MethodDefinition` came from. It happened to compile here, but it's still the wrong type.)